src/main/kotlin/io/easybreezy/project/api/controller/ProjectController.kt

```diff
@@ -3,27 +3,94 @@ package io.easybreezy.project.api.controller
 import com.google.inject.Inject
 import io.easybreezy.infrastructure.ktor.Controller
 import io.easybreezy.infrastructure.ktor.Response
-import io.easybreezy.project.application.project.command.Handler
-import io.easybreezy.project.application.project.command.New
-import io.easybreezy.project.application.project.command.NewRole
+import io.easybreezy.infrastructure.query.QueryExecutor
+import io.easybreezy.infrastructure.query.pagingParameters
+import io.easybreezy.infrastructure.structure.Either
+import io.easybreezy.project.application.project.command.*
+import io.easybreezy.project.application.project.queryobject.Project
+import io.easybreezy.project.application.project.queryobject.ProjectQO
+import io.easybreezy.project.application.project.queryobject.ProjectsQO
 import java.util.UUID
 
-class ProjectController @Inject constructor(private val handler: Handler) : Controller() {
-    suspend fun create(new: New): Response.Ok {
-        handler.new(new)
-        return Response.Ok
+class ProjectController @Inject constructor(
+    private val handler: Handler,
+    private val validation: Validation,
+    private val queryExecutor: QueryExecutor
+) : Controller() {
+
+    suspend fun create(new: New, author: UUID): Response.Either<Response.Ok, Response.Errors> {
+
+        val errors = validation.validate(new)
+        if (errors.isNotEmpty()) {
+            return Response.Either(Either.Right(Response.Errors(errors)))
+        }
+        handler.new(new, author)
```

Comment:
лучше author засунуть в команду и передавать только ее
`new.author = author`
самого author в команде пометить @Transient, чтобы сериализация не ругалась, есть примеры в проекте

Reply:
I did it in purpose because in my opinion author doesn't belong to command. He is an actor.

Reply:
Is it a reason that command shouldn't contain actor id?

Reply:
I don't know how exactly it should be. But technically the author is a required part of project creation.

Reply:
as I see the main problems in current realization that the handler method has 2 parameters, am I right?

Reply:
No, it's more philosophies question. Usually, we have command and his handler, in your variant, we have command and external parameters why are these parameters external? Why is it not a part of command? Because command usually has all information about feature action

Reply:
Because it is was different things on my mind, if we define command as an object with all parameters for action than ok I'll fix it and in other places too

Reply:
In our case we can look at command as DTO with strong contract, that contains all attributes to do some action in handler.